### PR TITLE
Beam model fix

### PR DESCRIPTION
**Pull request by keegansmith21**
Implementation of the FEE2016 beam model version for the beamformer

### CODE REVIEW

**Review by robotopia — APPROVED** [review]

File: make_beam/get_delays_small.c [hunk 1]
```diff
@@ -418,7 +442,12 @@ void get_delays(
 
                 // Now, calculate the inverse Jones matrix
                 if (invJi != NULL) {
-                    if (pol == 0) {
+                    if (pol == 0) { // This is just to avoid doing the same calculation twice
+                        // Apply parallactic angle correction if Hyperbeam was used
+                        //if (beam_model == BEAM_FEE2016) {
+                        //    mult2x2d_RxC( P, Ji, Ji );  // Ji = P x Ji (where 'x' is matrix multiplication)
+                        //}
```
Comment (NickSwainston) [inline make_beam/get_delays_small.c:449]: Why is this commented out? Might as well remove it if we don't need it

File: make_beam/get_delays_small.c [hunk 2]
```diff
@@ -428,16 +457,31 @@ void get_delays(
...
+                {
+                    for (p2 = 0; p2 < NPOL;  p2++)
+                    {
+                        fprintf(stderr, "%.18lf %.18lf ", CReald(invJi[5][ch][p1][p2]), CImagd(invJi[5][ch][p1][p2]));
+                    }
+                }
+                fprintf(stderr, "\n");
+            }
+        }
+        */
```
Comment (NickSwainston) [inline make_beam/get_delays_small.c:481]: Looks like this is just debugging, might want to remove it or label it as debugging

**Review by NickSwainston — COMMENTED** [review]
Just two minor comments. Before I pull I'd like some comparison plots of how the beamformer compares so we have some good bookkeeping in the code. I'd also like to update the automated docker build of the beamformer, does it need me to build this hyperbeam repo?

File: make_beam/get_delays_small.c (hunk 1 above)
Comment (robotopia) [reply make_beam/get_delays_small.c:449]: There _will_ have to be a parallactic angle correction at some point. Something about the way it's implemented here is wrong, hence why it's commented out, but leave it here as a memo to ourselves to figure out how to apply the correction, uh, correctly.

**Comment by robotopia** [comment]
Comparison plots? I'm working on a whole paper! But seriously, the differences are too complex to be summarised by just a few examples. Also, the FEE beam has already been verified to be pbjectively better (= closer to the truth) than the analytic beam by other groups, so I don't think showing some comparison plots are the right way (at this stage) of "convincing" a user that the FEE beam should be used.
For Hyperbeam, it's already installed on Garrawarla, but I don't think that answers your question. For a completely self-contained Docker image, I guess you _will_ have to build it...? Check with Chris -- maybe he has something already.

**Comment by keegansmith21** [comment]
> Just two minor comments. Before I pull I'd like some comparison plots of how the beamformer compares so we have some good bookkeeping in the code. I'd also like to update the automated docker build of the beamformer, does it need me to build this hyperbeam repo?

An updated docker image of the beamformer will require hyperbeam because this is the common codebase we decided upon using for implementation of the FEE beam

File: make_beam/get_delays_small.c (hunk 1 above)
Comment (NickSwainston) [reply make_beam/get_delays_small.c:449]: O I see, you haven't got to that bit yet.

**Comment by NickSwainston** [comment]
> Comparison plots? I'm working on a whole paper! But seriously, the differences are too complex to be summarised by just a few examples. Also, the FEE beam has already been verified to be pbjectively better (= closer to the truth) than the analytic beam by other groups, so I don't think showing some comparison plots are the right way (at this stage) of "convincing" a user that the FEE beam should be used.
That makes sense. Once it's published, remind me to put the paper on the readme

Alright I'll message Chris and once I've got the docker image working I'll merge the branch

**Review by NickSwainston — APPROVED** [review]

**Pull request merged**